src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -134,5 +127,268 @@ private BoundExpression BindInterpolatedString(InterpolatedStringExpressionSynta
+                // a placeholder, and put the part into the interpolated string's list of parts.
+                Debug.Assert(builderArguments.Length == unconvertedInterpolatedString.Parts.Length);
+                var interpolatedDataBuilder = ArrayBuilder<MethodArgumentInfo>.GetInstance(builderArguments.Length);
+                var argumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(3);
+                var partsBuilder = ArrayBuilder<BoundExpression>.GetInstance(builderArguments.Length);
+                int baseStringLength = 0;
+                int numFormatHoles = 0;
+
+                for (int i = 0; i < builderArguments.Length; i++)
+                {
+                    var currentFormatCall = builderArguments[i];
+                    Debug.Assert(currentFormatCall.Arguments.Length > 0);
+                    Debug.Assert(argumentsBuilder.Count == 0);
+
+                    var newPart = currentFormatCall.Arguments[0];
+                    argumentsBuilder.Add(new BoundInterpolatedStringElementPlaceholder(newPart.Syntax, i, newPart.Type));
+                    if (currentFormatCall.Arguments.Length > 1)
+                    {
+                        argumentsBuilder.AddRange(currentFormatCall.Arguments, 1, currentFormatCall.Arguments.Length - 1);
+                    }
+
+                    interpolatedDataBuilder.Add(currentFormatCall with { Arguments = argumentsBuilder.ToImmutableAndClear() });
+
+                    var currentPart = unconvertedInterpolatedString.Parts[i];
+                    if (currentPart is BoundStringInsert insert)
+                    {
+                        numFormatHoles++;
+
+                        if (needToCheckConversionToObject)
+                        {
+                            Debug.Assert(conversionDiagnostics is not null);
+                            var value = insert.Value;
+                            bool reported = false;
+                            if (value.Type is not null)
+                            {
+                                value = BindToNaturalType(value, conversionDiagnostics);
+                                if (conversionDiagnostics.HasAnyErrors())
+                                {
+                                    CheckFeatureAvailability(value.Syntax, MessageID.IDS_FeatureImprovedInterpolatedStrings, diagnostics);
+                                    conversionDiagnostics.Clear();
+                                    reported = true;
+                                }
+                            }
+
+                            if (!reported)
+                            {
+                                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(conversionDiagnostics);
+                                var conversion = Conversions.ClassifyConversionFromExpression(value, objectType, ref useSiteInfo);
```

Comment:
>Conversions.ClassifyConversionFromExpression(value, objectType, ref useSiteInfo); [](start = 49, length = 81)

Is this sufficient to pick up any errors that could come from ```GenerateConversionForAssignment```. Consider using ```GenerateConversionForAssignment``` to match what legacy compiler would do.